packages/runtime/src/interpreter/opcode-list.ts

```diff
@@ -2591,3 +2591,84 @@ export class MinBalance extends Op {
     stack.push(BigInt(acc.minBalance));
   }
 }
+
+// push Ith scratch space index of the Tth transaction in the current group
+// push to stack [...stack, bigint/bytes]
+// Pops nothing
+// Args expected: [{uint8 transaction group index}(T),
+// {uint8 position in scratch space to load from}(I)]
+export class Gload extends Op {
+  readonly scratchIndex: number;
+  readonly txIndex: number;
+  readonly interpreter: Interpreter;
+  readonly line: number;
+
+  /**
+   * Stores scratch space index and transaction index number according to arguments passed.
+   * @param args Expected arguments: [index number]
+   * @param line line number in TEAL file
+   * @param interpreter interpreter object
+   */
+  constructor (args: string[], line: number, interpreter: Interpreter) {
+    super();
+    this.line = line;
+    assertLen(args.length, 2, this.line);
+    assertOnlyDigits(args[0], this.line);
+    assertOnlyDigits(args[1], this.line);
+
+    this.txIndex = Number(args[0]);
+    this.scratchIndex = Number(args[1]);
+    this.interpreter = interpreter;
+  }
+
+  execute (stack: TEALStack): void {
+    const scratch = this.interpreter.runtime.ctx.sharedScratchSpace.get(this.txIndex);
+    if (scratch === undefined) {
+      throw new RuntimeError(
+        RUNTIME_ERRORS.TEAL.SCRATCH_EXIST_ERROR,
+        { index: this.txIndex, line: this.line }
+      );
+    }
```

Comment:
just to confirm, Is this validated on node?

Reply:
it will be with integration tests. I have been waiting for this one's review.